docs/data/data-grid/editing-legacy/editing.md

```diff
@@ -0,0 +1,272 @@
+Once the value is validated in the server, the promise should be resolved with a new object containing the `error` attribute set to true or false.
+The grid will wait for the promise to be resolved before exiting the edit mode.
+
+<!-- TODO v6: make the following paragraph the default behavior -->
+
+> By default, `preProcessEditCellProps` is called on each value change and during commit.
+> With column types that automatically commit the value after a selection (e.g. `singleSelect`) this means that the callback will be called twice and with the wrong value in the second call.
+> To avoid this inconsistency, enable the `preventCommitWhileValidating` flag available in the experimental features.
+>
+> ```jsx
+> <DataGrid experimentalFeatures={{ preventCommitWhileValidating: true }} />
+> ```
+>
+> It's labeled as experimental because in v6 this flag will become the default behavior.
+
+This demo shows how you can validate a username asynchronously and prevent the user from committing the value while validating.
+It's using `DataGridPro` but the same approach can be used with `DataGrid`.
+
+{{"demo": "ValidateServerNameGrid.js", "bg": "inline", "defaultCodeOpen": false}}
+
+### Server-side persistence
+
+If you want to send the updated data to your server, you can use the `onCellEditCommit` which is fired just before committing the new cell value to the grid.
+
+You can then decide if you want to send the whole row or only the modified fields.
+
+{{"demo": "CellEditServerSidePersistence.js", "bg": "inline", "defaultCodeOpen": false}}
+
+### Custom edit component
+
+To customize the edit component of a column, use the `renderEditCell` attribute available in the `GridColDef`.
+
+The demo lets you edit the ratings by double-clicking the cell.
+
+{{"demo": "RenderRatingEditCellGrid.js", "bg": "inline", "defaultCodeOpen": false}}
+
+### Edit using external button [<span class="plan-pro"></span>](https://mui.com/store/items/material-ui-pro/)
+
+You can override the default [start editing](#start-editing) triggers using the [`event.defaultMuiPrevented`](/components/data-grid/events#disabling-the-default-behavior) on the synthetic React events.
+
+{{"demo": "StartEditButtonGrid.js", "bg": "inline", "disableAd": true}}
+
+### Events [<span class="plan-pro"></span>](https://mui.com/store/items/material-ui-pro/)
+
+The editing feature leverages the event capability of the grid and the apiRef.
+The following events can be imported and used to customize the edition:
+
+- `cellEditStart`: emitted when the cell turns to edit mode.
+- `cellEditStop`: emitted when the cell turns back to view mode.
+- `cellEditCommit`: emitted when the new value is committed.
+- `editCellPropsChange`: emitted when the props passed to the edit cell component are changed.
+
+Catching events can be used to add a callback after an event while ignoring its triggers.
+
+The demo shows how to catch the start & end edit events to log which cell is editing in an info message:
```

Comment:
Let's spell out "and" instead of using an ampersand. Also, I think the verb conjugation we want in the latter half of the sentence is "has been edited," but I could be wrong if I've misunderstood the context. 

```suggestion
The demo shows how to catch the start and end edit events in order to log which cell has been edited in an info message:
```

Reply:
Unless it's more accurate to use the plural, in which case we'd want to say "which cells have been edited."

Reply:
The demo only logs the current cell so the singular version is more correct.